### DISCUSSION TITLE
No data from HMS

### DISCUSSION THREAD

**Discussion by Wiki591** [DennisOSRM/hms-mqtt-publisher#29] [General]
Hi,
at first Thank You for sharing this project.

I've compiled the sources using RaspberryPi, Ubuntu 20.04 and cargo/rust 1.73.0 for a friend. Starting the binary with given ip-adresses of inverter and mqtt broker tells me, that the inverter is online. But nothing else happens, no data are published. The inverter is a HMS 800W-T2, checking the version in the hoymiles app it tells V1.1.15, I suppose its the version number of the DTU.

What I didn't do was setting the environment variables.

Any ideas?

[edit]
additional: normally I am connection anonymous to the broker, so I didn't deliver neither mqttuser/pwd nor portno. to the tool
[/edit]

**Comment by T3mplate-wow** [comment]
May you have the same problem like I had. 
I thought that there are no data’s incoming, but there is just no auto discovery. That means no entity will be there automatically in HA or ioBroker or whatever. 
you have to fetch the data with a manual device.

**Comment by Wiki591** [comment]
Erm, I am trying to build a bridge to domoticz using a small python script, subscribing and listening for topic hms800wt2/#

mosquitto_sub -t 'hms800wt2/#' -h [broker-ip] gives no data at all.

**Comment by Wiki591** [comment]
Ähm, the version I posted is the version of the app which my friend sent to me, sorry. I asked him once again, inverter software version V1.00.08, HW H00.04.00, DTU software V00.01.08, HW H00.01.00

**Comment by Wiki591** [comment]
OK, after some time of reading on discord I get more transparency in my issue. I tried to connect and read the data from the hms using the existing wlan, the inverter connected to a fritzbox. Did I learn right, that is not the way the tools is designed for?

Is there somewhere a howto which summarizes the prerequesites to collect the data from the inverter?

**Comment by DennisOSRM** [comment]
If you get the message that the inverter is online, then we have received a valid response with the current readings from it. In other words, it's actually reachable on the network. This would point to the MQTT broker not being reached as expected.

If you set the `RUST_LOG=debug` environment variable, then you should get (much) more detailed logging.

**Comment by Wiki591** [comment]
Do I have to set RUST_LOG during compilation or would it be sufficient to set it before starting the binary?

My work is a little bit complicate because my friend owns a raspi 3, with which compiling the tool is really no fun. So I am compiling here at home and carrying the binary with me on my next visit.

**Comment by DennisOSRM** [comment]
The environment variable is read at runtime. There is no need to recompile, and setting it at start of the binary should be enough.

**Comment by Wiki591** [comment]
I appreciate your support. Reporting any results will take some days.

**Comment by Wiki591** [comment]
Success!
I compiled the last version today in the morning and went over to my friend, filled the conf-file, started the binary and listened for topics 'hms800wt2/#'. And yes, there are data published as lean mqtt topics. Great, Thanks for your work and sharing it!

Questions: 

- I commented out the part for Home Assistant in the conf file. Starting the binary I only get the info 'Publishing to simple MQTT broker', the info 'Publishing to Home Assistant' didn't appear. Thats what I wanted. Having a look at the mqtt traffic there still appear the data published for Home Assistant. Is it somehow possible to avoid this not needed traffic, just to get the lean 'hms800wt2' topics?
- As I noticed, the amount of information raised. Will it be possible in the future to get the information of the overall produced energy, something like pv_total_yield?

[edit]
fyi: Setting the environment variable RUST_LOG=debug didn't produce any additional output
[/edit]

**Comment by DennisOSRM** [comment]
> Success! I compiled the last version today in the morning and went over to my friend, filled the conf-file, started the binary and listened for topics 'hms800wt2/#'. And yes, there are data published as lean mqtt topics. Great, Thanks for your work and sharing it!

👍 

> Questions:
> 
> * I commented out the part for Home Assistant in the conf file. Starting the binary I only get the info 'Publishing to simple MQTT broker', the info 'Publishing to Home Assistant' didn't appear. Thats what I wanted. Having a look at the mqtt traffic there still appear the data published for Home Assistant. Is it somehow possible to avoid this not needed traffic, just to get the lean 'hms800wt2' topics?

Removing the undesired section from the config file should stop publishing the data.

> * As I noticed, the amount of information raised. Will it be possible in the future to get the information of the overall produced energy, something like pv_total_yield?

This has been added in main branch by now.

> [edit] fyi: Setting the environment variable RUST_LOG=debug didn't produce any additional output [/edit]

That's a known issue that was fixed very recently in #58

**Comment by Wiki591** [comment]
> > * As I noticed, the amount of information raised. Will it be possible in the future to get the information of the overall produced energy, something like pv_total_yield?
> 
> This has been added in main branch by now.
> 
Erm, looking at the sources for simple mqtt I can't see any topic publishing the overall produced energy of the inverter, only for the different ports.

Did I miss something?

**Reply by DennisOSRM** [reply]
IIRC, this isn't published as a total by the inverter. Not sure why. FWIW, I am adding the two port-based values in my own dashboards.

**Reply by Wiki591** [reply]
Ah, OK, so I will follow your way. Thank you.

Have a happy New Year!

**Comment by KBH10** [comment]
Hi together,

I would like to join this conversation. I also want to use the hms-mqtt-publisher in iobroker.
It is working fine so far. The adapter is started correctly. When i listen to the topics using mosquitto_sub, e.g, it is working fine, I see publishing topics every 30.50 sec.. But it seems only working, when i use the [home assistant] config in the config.toml file.

It seems also somehow working with iobroker, that it got updated. But not in extra states, like:

State: PV_1_Power: 170
State: PV_2_Power: 150
State: PV_Current_Power: 320

Its all in one file, like:
State:
{
   " .... ",
  "pv_1_power": "190.50",
  "pv_2_power": "195.30",
  "pv_current_power": "365.50",
}

Is it possible to get it somehow in extra states? Or do i have to extract it on my own, with a little script?

Many Thanks in advance for the replay.
And of course also many thanks for this nice possibility to get the info of my hms800wt2 directly in my iobroker.

Best regards,
KBH10
